.github/workflows/typespec-requirement.yaml

```diff
@@ -0,0 +1,32 @@
+name: TypeSpec Requirement
+
+on: pull_request
+
+jobs:
+  TypeSpec-Requirement:
+    strategy:
+      fail-fast: false
+      matrix:
+        spec-type: [data-plane, resource-manager]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
```

Comment:
Why depth 2?

Reply:
Default is depth 1 for perf.  However, TypeSpecRequirement needs to compute the `git diff` against `HEAD^`, which requires depth of at least two (current commit and previous commit).

Reply:
I added comments and parameters making this more clear